Add config options to disable footnote anchor prefix and/or header ID suffix

jacobwgillespie opened the pull request:
The new config options are

* `DisableFootnoteAnchorPrefix` - bool - default: `false`
* `DisableHeaderIDSuffix` - bool - default: `false`

My site depends on anchor IDs being generated without the suffix, so I added this configuration option to disable the automatic suffix behavior.  I understand the risk of duplicate IDs, so that's why this is defaulting to `false`.

So, if the config file contains:

```toml
DisableHeaderIDSuffix = true
```

Then this:

```markdown
# My Header
```

will generate:

```html
<h1 id="my-header">My Header</h1>
```

bjornerik commented:
Not sure, but these looks like Blackfriday options - and should maybe be grouped with the others. Agree, @halostatue ?

See #744

halostatue commented:
Agree. Checking for `len(ctx.DocumentId) != 0` should probably be done once, though, not twice as it is now.

halostatue commented:
I’m also wondering if this is not really one meta-option that controls both of these. (That is, do we need two knobs or just one? I can’t see where you’d just turn off footnote anchor prefixes but not document anchor suffixes.)

jacobwgillespie commented:
Yeah, that makes more sense - I'm not sure the use case for turning off one or the other.  Let me adapt to be a Blackfriday option for both and I'll push.

jacobwgillespie commented:
Okay, I renamed it to Blackfriday: documentIDAnchor.  Not exactly happy with that name, but it's what I could come up with at the moment.

Also added some documentation.

jacobwgillespie commented:
I broke the tests - one sec...

jacobwgillespie commented:
Okay - renamed it to `plainIdAnchors`, default `false`, which seems more logical.  Should be ready for review.
